modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -174,8 +177,63 @@ object SeqexecEngine {
      * TODO: Implement. I suspect this needs to return F[Boolean]
      * @return true if target is valid
      */
-    private def sequenceTcsTargetMatch: Boolean =
-      true
+    private def sequenceTcsTargetMatch(obs: SequenceData[F], stp: Option[StepId]): F[Option[(StepId, String, String)]] = (
+      for {
+        stepId    <- stp.orElse(obs.seq.currentStep.map(_.id))
+        a         <- obs.seqGen.steps.dropWhile(_.id =!= stepId).find(a => stepRequiresTargetCheck(a.config))
+        seqTarget <- extractTargetName(a.config)
+      } yield systems.tcsKeywordReader.sourceATarget.objectName.map { tcsTarget =>
+        (seqTarget =!= tcsTarget).option((stepId, seqTarget, tcsTarget))
+      }
+    ).getOrElse(none.pure[F])
+//      val stepId = stp.orElse(obs.seq.currentStep.map(_.id)) // start from either the given step or the first not run
+//      val seqTargetO = stepId
+//        .flatMap(x => obs.seqGen.steps.dropWhile(_.id =!= x).find(a => stepRequiresTargetCheck(a.config)))
+//        .flatMap(x => extractTargetName(x.config))
+//
+//      systems.tcsKeywordReader.sourceATarget.objectName.map( tcsTarget =>
+//        seqTargetO.flatMap( seqTarget =>
+//          (seqTarget =!= tcsTarget).option((seqTarget, tcsTarget))
+//        )
+//      )
+//    }
+
+    /**
+     * Extract the target name from a step configuration. Some processing is necessary to get the same string that
+     * appears in TCS.
+     */
+    private def extractTargetName(config: CleanConfig): Option[String] = {
+      val BasePositionKey = "Base:name"
+      val SolarSystemObjects = Map(
```

Comment:
Should this go into an `Enumeration`?

Reply:
It is a translation table, I don't see any gain on making it an `Enumeration`.